### PR TITLE
Fullscreen WebEntity and magic window

### PR DESCRIPTION
**Pull request by LeonCrew**
Full screen take over of any web entity clicked

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
An admin must verify this patch for the pull request build to fire. Type "ok to test" to begin testing this pull request or "add to whitelist" to test this and all future pull requests from this collaborator.

**Comment by themelissabrown** [comment]
add to whitelist

**Comment by hifi-gustavo** [comment]
Build console output for commit: 0d7acf02d915aab1a52dbb6f7eb9f37f0457f938 available at: https://gist.github.com/387bbda2e1c13a1c07c80eae03e4f665<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10434-0d7acf0.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10434-0d7acf0.dmg

**Comment by LeonCrew** [comment]
Add magic window functionality

**Comment by hifi-gustavo** [comment]
Build console output for commit: 20179ebdc14241bb5272d87a652edc0ab994468a available at: https://gist.github.com/387bbda2e1c13a1c07c80eae03e4f665<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10434-20179eb.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10434-20179eb.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: db0eff1e07f814739f3fd32e8f23ec83366481b3 available at: https://gist.github.com/387bbda2e1c13a1c07c80eae03e4f665<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10434-db0eff1.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10434-db0eff1.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 643a39e02e9a558e28fe2e57cd2fad023d8014ac available at: https://gist.github.com/387bbda2e1c13a1c07c80eae03e4f665<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10434-643a39e.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10434-643a39e.dmg

**Comment by sethalves** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 643a39e02e9a558e28fe2e57cd2fad023d8014ac available at: https://gist.github.com/387bbda2e1c13a1c07c80eae03e4f665<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10434-643a39e.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10434-643a39e.dmg

File: interface/src/Application.cpp [hunk 1]
```diff
@@ -3523,6 +3525,16 @@ void Application::touchGestureEvent(QGestureEvent* event) {
     if (_touchscreenDevice && _touchscreenDevice->isActive()) {
         _touchscreenDevice->touchGestureEvent(event);
     }
+    if (QGesture* swipe = event->gesture(Qt::SwipeGesture))	{
+        QSwipeGesture* swipeGesture = static_cast<QSwipeGesture*>(swipe);
+        if (swipeGesture->horizontalDirection() == QSwipeGesture::Left)	{
+            auto avatarManager = DependencyManager::get<AvatarManager>();
+            avatarManager->getMyAvatar()->getHead()->setOrientation(getMyAvatar()->getHead()->getOrientation() * glm::quat(glm::radians(glm::vec3(0.0f, -15.0f, 0.0f))));
```
Comment (AndrewMeadows) [inline interface/src/Application.cpp]: The `15.0f` is considered a "magic number": a literal value that doesn't have a name or obvious meaning.  At the very least that needs to be given a name.  See the **coding standard section 3.5.1**:
https://github.com/highfidelity/hifi/wiki/Coding-Standard#351-constants-and-magic-numbers

Meanwhile, I would recommend making the code a bit more readable by doing something like this:
```
const float SWIPE_ANGLE = glm::radians(-15.0f);
auto head = DependencyManager::get<AvatarManager>()->getMyAvatar()->getHead();
auto swipeDirection = swipeGesture->horizontalDirection();
if (swipeDirection == QSwipeGesture::Left) {
    head->setOrientation(head->getOrientation() * glm::angleAxis(-SWIPE_ANGLE, UNIT_Y);
} else if (swipeDirection == QSwipeGesture::Right) {
    head->setOrientation(head->getOrientation() * glm::angleAxis(SWIPE_ANGLE, UNIT_Y);
}
```

File: libraries/entities-renderer/src/RenderableWebEntityItem.cpp [hunk 2]
```diff
@@ -359,6 +361,30 @@ void RenderableWebEntityItem::handlePointerEvent(const PointerEvent& event) {
...
+            
+            auto offscreenUi = DependencyManager::get<OffscreenUi>();
+            QString infoViewName(NAME);
+            offscreenUi->show(QML, NAME, [=](QQmlContext* context, QObject* newObject){
+                QQuickItem* item = dynamic_cast<QQuickItem*>(newObject);
+                
+                /* Use base window size
+                Application *app = static_cast<Application*>(QCoreApplication::instance());
+                QSize windowSize = app->getPrimaryWindow()->size();*/
+                item->setSize(QSize(1920, 1080));
```
Comment (AndrewMeadows) [inline libraries/entities-renderer/src/RenderableWebEntityItem.cpp]: I don't think you want to hard code `1920x1080` here.  What if someone is using a differently sized monitor?  Was this debug code for a personal setup or... something else?

If this new code really is what you want then you should delete the commented out block above it for cruft.

File: libraries/entities-renderer/src/RenderableWebEntityItem.cpp [hunk 3]
```diff
@@ -359,6 +361,30 @@ void RenderableWebEntityItem::handlePointerEvent(const PointerEvent& event) {
             break;
         }
 
+        if (event.getType() == PointerEvent::Press)
+        {
```
Comment (AndrewMeadows) [inline libraries/entities-renderer/src/RenderableWebEntityItem.cpp]: Formatting is wrong here according to **coding standard section 4.1.2**:
https://github.com/highfidelity/hifi/wiki/Coding-Standard#412-use-inline-braces-for-block-layout

**Comment by hifi-gustavo** [comment]
Build console output for commit: bff18d8aeb675e675213cf2e75c324c792eb0cad available at: https://gist.github.com/387bbda2e1c13a1c07c80eae03e4f665<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10434-bff18d8.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10434-bff18d8.dmg

**Pull request closed without merging**